### PR TITLE
feat(runes): add Get Tokens API

### PR DESCRIPTION
**Pull request by gazenw**
## Description
- Implement Get Tokens API
  - Supports 2 scopes (`all` and `ongoing`)
    - `all` (DEFAULT) returns all rune entries, sorted by etching order
    - `ongoing` returns rune entries where mints < cap, sorted by mint progress (mint / cap)
  - Supports a search string, which does a prefix search on the rune name (spacers are trimed).
- Refactor other endpoints with pagination to use `paginationRequest` struct
- Add a GIN index on `runes_entries.rune` column to speed up `LIKE` operator.

## Type of change

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (improvement to existing features and functionality)
- [ ] Documentation update (changes to documentation)
- [ ] Performance improvement (non-breaking change which improves efficiency)
- [ ] Code consistency (non-breaking change which improves code reliability and robustness)

## Commit formatting

Please follow the commit message conventions for an easy way to identify the purpose or intention of a commit. Check out our commit message conventions in the [CONTRIBUTING.md](https://github.com/gaze-network/gaze-indexer/blob/main/.github/CONTRIBUTING.md#pull-requests-or-commits)

### CODE REVIEW

File: modules/runes/database/postgresql/queries/data.sql [hunk 1]
```diff
@@ -13,6 +13,9 @@ SELECT * FROM balances WHERE amount > 0 ORDER BY amount DESC, pkscript LIMIT $3
 -- name: GetBalanceByPkScriptAndRuneId :one
 SELECT * FROM runes_balances WHERE pkscript = $1 AND rune_id = $2 AND block_height <= $3 ORDER BY block_height DESC LIMIT 1;
 
+-- name: GetTotalHoldersByRuneIds :many
+SELECT rune_id, COUNT(pkscript) FROM runes_balances WHERE rune_id = ANY(@rune_ids::TEXT[]) AND amount > 0 GROUP BY rune_id;
+
```
Comment (bombnp) [inline modules/runes/database/postgresql/queries/data.sql:18]: count only latest balances for each (runeId, pkScript)

**Review by bombnp — CHANGES_REQUESTED** [review]

File: modules/runes/api/httphandler/get_token_list.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	blockHeight := req.BlockHeight
+	if blockHeight == 0 {
+		blockHeader, err := h.usecase.GetLatestBlock(ctx.UserContext())
+		if err != nil {
+			if errors.Is(err, errs.NotFound) {
+				return errs.NewPublicError("latest block not found")
+			}
+			return errors.Wrap(err, "error during GetLatestBlock")
+		}
+		blockHeight = uint64(blockHeader.Height)
```
Comment (Planxnx) [inline modules/runes/api/httphandler/get_token_list.go]:
```suggestion
		//nolint:gosec 
		blockHeight = uint64(blockHeader.Height)
```

File: modules/runes/api/httphandler/get_token_list.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		if err != nil {
+			return errors.Wrap(err, "error during GetRuneEntryList")
+		}
+	case GetTokensScopeMinting:
+		entries, err = h.usecase.GetMintingRuneEntries(ctx.UserContext(), search, blockHeight, req.Limit, req.Offset)
+		if err != nil {
+			return errors.Wrap(err, "error during GetRuneEntryList")
+		}
+	default:
+		panic("invalid scope")
```
Comment (Planxnx) [inline modules/runes/api/httphandler/get_token_list.go]: shouldn't panic in http handler, do return `errs.InternError` or something instead

File: modules/runes/api/httphandler/get_token_list.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		result = append(result, getTokenInfoResult{
+			Id:                ent.RuneId,
+			Name:              ent.SpacedRune,
+			Symbol:            string(ent.Symbol),
+			TotalSupply:       totalSupply,
+			CirculatingSupply: circulatingSupply,
+			MintedAmount:      mintedAmount,
+			BurnedAmount:      ent.BurnedAmount,
+			Decimals:          ent.Divisibility,
+			DeployedAt:        uint64(ent.EtchedAt.Unix()),
```
Comment (Planxnx) [inline modules/runes/api/httphandler/get_token_list.go]: use `int64` for unix timestamp value

**Review by Planxnx — CHANGES_REQUESTED** [review]

**Review by Planxnx — APPROVED** [review]

**Pull request merged**